src/Features/CSharp/Portable/Completion/CompletionProviders/CSharpSuggestionModeCompletionProvider.cs

```diff
@@ -146,17 +146,17 @@ private static bool IsLambdaExpression(SemanticModel semanticModel, SyntaxTree t
             // Walk up a single level to allow for typing the beginning of a lambda:
             // new AssemblyLoadEventHandler(($$
             if (token.IsKind(SyntaxKind.OpenParenToken) &&
-                token.GetRequiredParent().IsKind(SyntaxKind.ParenthesizedExpression))
+                token.Parent.IsKind(SyntaxKind.ParenthesizedExpression))
```

Comment:
This is not quite the same right? This will silently handle nulls and return false, but previously it would have been a contract violation. Couldn't this introduce undefined behaviour?

(same goes for other `GetRequiredParent()` calls that are being removed

Reply:
It's slightly different, but `GetRequiredParent()` was used here as an indication that `token.Parent` is not expected to be null at this location.